web/client/api/Model.js

```diff
@@ -0,0 +1,228 @@
+                eastings: mapConversionObj?.Eastings?.value,            // y coord
+                orthogonalHeight: mapConversionObj?.OrthogonalHeight?.value,    // height (z coord)
+                xAxisOrdinate: mapConversionObj?.XAxisOrdinate?.value,
+                xAxisAbscissa: mapConversionObj?.XAxisAbscissa?.value,
+                rotation: Math.atan2(mapConversionObj?.XAxisOrdinate?.value || 0, mapConversionObj?.XAxisAbscissa?.value || 0) * 180.0 / Math. PI,
+                scale: mapConversionObj?.Scale?.value
+            };
+            if (proj4.defs(projectedCrs)) {         // if crs in not defined in MS, model will be locatied at 0,0
+                let wgs84Origin = proj4(proj4.defs(projectedCrs), proj4.defs('EPSG:4326'), [mapConversion.eastings, mapConversion.northings]);
+                originProperties = {
+                    projectedCrs,
+                    longitude: wgs84Origin[0] || 0,
+                    latitude: wgs84Origin[1] || 0,
+                    height: mapConversion.orthogonalHeight || 0,
+                    scale: mapConversion.scale || 1,
+                    heading: mapConversion.rotation,
+                    mapConversion
+                };
+            } else {
+                originProperties = {
+                    projectedCrs,
+                    projectedCrsNotSupported: true,
+                    mapConversion
+                };
+            }
+        }
+    }
+    // todo: maybe in future enhancement, handling georeferenced ifc models with schema less than version 4 like IFC2x3 by getting projection info ref: (https://medium.com/@stijngoedertier/how-to-georeference-a-bim-model-1905d5154cfd)
+    return originProperties;
+}
+
+// extract the tile format from the uri
+function getFormat(uri) {
+    const parts = uri.split(/\./g);
+    const format = parts[parts.length - 1];
+    return format;
+}
+
+// extract version, bbox, format and properties from the ifc file
+function extractCapabilities(ifcData, modelID, url) {
+    const { ifcApi } = ifcData;
+    const version = ifcApi?.GetModelSchema(modelID) !== undefined ? ifcApi.GetModelSchema(modelID)?.includes("IFC4")? "IFC4" : ifcApi.GetModelSchema(modelID) : 'IFC4';    // eslint-disable-line
+    const format = getFormat(url || '');
+    return {
+        version,
+        format,
+        properties: {}
+    };
+}
+
+/**
+ * get ifc response and additional parsed information such as: version, bbox, format and properties
+ * @param {object} this object includes: data --> ifc raw data, ifcData: is an object includes ifcApi object (from web-ifc) that enable to read ifc content
+ * @return {object} return json object includes meshes array [geometry data of ifc model], extent of ifc model, center and size
+ * @
+ */
+export const ifcDataToJSON = ({ data, ifcData }) => {
+    const { ifcApi } = ifcData;
+    const settings = {
+        COORDINATE_TO_ORIGIN: false, // this property change the position for IFC4 with projection if true
```

Comment:
After some tests I noticed this property set to false change the position of projected IFC4 model, please double check if this is working also for non projected model